src/platform_impl/linux/wayland/window/state.rs

```diff
@@ -269,24 +269,26 @@ impl WindowState {
             && self.frame.is_none()
             && !self.csd_fails
         {
-            match WinitFrame::new(
-                &self.window,
-                shm,
-                subcompositor.clone(),
-                self.queue_handle.clone(),
-                #[cfg(feature = "sctk-adwaita")]
-                into_sctk_adwaita_config(self.theme),
-            ) {
-                Ok(mut frame) => {
-                    frame.set_title(&self.title);
-                    frame.set_scaling_factor(self.scale_factor);
-                    // Hide the frame if we were asked to not decorate.
-                    frame.set_hidden(!self.decorate);
-                    self.frame = Some(frame);
-                }
-                Err(err) => {
-                    warn!("Failed to create client side decorations frame: {err}");
-                    self.csd_fails = true;
+            if let Some(subcompositor) = subcompositor {
```

Comment:
Can we embed the check for subcompositor into the `if`, so we can unwrap in the code below reducing the nesting?

I just don't like the way we add more nesting to this, where it was already hard...

Reply:
Alternative is of course an `if subcompositor.is_some()` and unwrapping in the body...